src/template/value.ts

```diff
@@ -13,13 +13,12 @@ export class Value {
   /**
    * @param str the value to be valuated, eg.: "foobar" | truncate: 3
    */
-  public constructor (input: string | FilteredValueToken, liquid: Liquid) {
-    const token: FilteredValueToken = typeof input === 'string'
-      ? new Tokenizer(input, liquid.options.operators).readFilteredValue()
-      : input
+  public constructor (input: string | FilteredValueToken | TagToken, liquid: Liquid) {
+    const token = this.getToken(input, liquid)
```

Comment:
Update: sorry I just find `tagToken.tokenizer` maybe already the tokenizer we want to expose with `argsTokenizer`, could you try to use it instead of constructing a new one?

Reply:
👍 That's much better.